fakeserver/main.go

```diff
@@ -86,6 +86,29 @@ func (s *server) SendKubernetesResources(stream pb.KubernetesInfoService_SendKub
 	}
 }
 
+func (s *server) KubernetesLogs(stream pb.KubernetesInfoService_KubernetesLogsServer) error {
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		panic("Failed to build zap logger: " + err.Error())
+	}
+	for {
+		req, err := stream.Recv()
+		if err == io.EOF {
+			// The client has closed the stream
+			logger.Info("Server has closed the stream")
+			return nil
+		}
+		if err != nil {
+			logger.Error("Error from stream", zap.Error(err))
+			return err // Return the error to terminate the stream
+		}
+
+		logs := req.Logs
+		logger.Info("Log messages")
+		logger.Info(logs)
```

Comment:
1. Log all the fields from the protobuf.
2. Use the correct log level from the received message.
3. Assume that the text message is JSON, decode it into a `json.RawMessage`, and log it as a structured `Object` field.
And you should do the same in the real server.
```suggestion
		var level zapcore.Level = ... // Map from the log level from the protobuf message's log level
		var message json.RawMessage // Decode the protobuf message's message string from JSON into this RawMessage...
		if ce := logger.Check(level, "Received log from cloud-operator"); ce != nil {
			ce.Write(
				zap.String("cluster_id", ...),
				zap.Object("message", message),
			)
		}
```

Reply:
Hmmm calling `zap.Object` with a `json.RawMessage` needs to be tested, but it should be possible to nest the message as a JSON object.